### PR TITLE
Auto-registration support

### PR DESCRIPTION
**Pull request by decarufe**
Fix last change to support automatic registration and add some method built-in to do it.

### CODE REVIEW

**Comment by decarufe** [comment]
Now support 3 ways to auto add handlers

By the initial mediator creation
`services.AddMediator(<list of assemblies to scan>)`

After mediator creation
`services.AddHandler(<list of assemblies to scan>)`
`services.AddHandler(<filter function of types to include>, <list of assemblies to scan>)`

File: src/GSoft.Cqrs/Configuration/DependencyInjection/BuilderExtensions/CqrsBuilderExtensions.cs [hunk 1]
```diff
@@ -13,6 +14,45 @@ public static void AddMediator(this IServiceCollection services)
         services.AddSingleton<RegistrationCollection>();
     }
 
+    public static void AddMediator(this IServiceCollection services, params Assembly[] assemblyToScan)
```
Comment (felpel) [inline src/GSoft.Cqrs/Configuration/DependencyInjection/BuilderExtensions/CqrsBuilderExtensions.cs]:
```suggestion
    public static void AddMediator(this IServiceCollection services, params Assembly[] assembliesToScan)
```

**Comment by felpel** [comment]
LGTM 👍

File: src/GSoft.Cqrs.Tests/CqrsBuilderExtensionsTests.cs [hunk 2]
```diff
@@ -101,6 +109,42 @@ public async Task AddHandler_Can_Register_Command_Handlers_And_Query_Handlers()
         Assert.Equal("rabbles", res3);
     }
 
+    [SuppressMessage("StyleCop.CSharp.OrderingRules", "SA1201:ElementsMustAppearInTheCorrectOrder", Justification = "Getting this close to its usage.")]
```
Comment (lballaire) [inline src/GSoft.Cqrs.Tests/CqrsBuilderExtensionsTests.cs]: I understand why you put it there, but I would recommend following the rules to ensure a better consistence accross classes.

I feel like it's easier to have those fields that can be reused across multiple methods at the top. It gives better visibility of what's already available.

**Review by lballaire — APPROVED** [review]

File: src/GSoft.Cqrs.Tests/CqrsBuilderExtensionsTests.cs (hunk 2 above)
Comment (decarufe) [reply src/GSoft.Cqrs.Tests/CqrsBuilderExtensionsTests.cs]: J'ai hésité. Je l'ai mis là pour la proximité et parce que l'utilisation est limité à quelques méthode. Mais tu as raison je vais les déplacer en haut.

**Review by felpel — APPROVED** [review]

**Pull request merged**